Jenkinsfile

```diff
@@ -165,10 +177,51 @@ pipeline {
                 }
             }
         }
+        stage('Retag previous commit images if possible'){
+            when {
+                allOf {
+                    expression { FILES_CHANGED == null }
+                    expression { IMAGE_EXISTS == null }
+                    expression { PREVIOUS_IMAGE_EXISTS != null }
+                }
+            }
+            steps {
+                withCredentials([usernamePassword(credentialsId: 'hub.docker.com', passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+                    sh '''
+                        URI_BASE=perconalab/percona-postgresql-operator:$VERSION
+                        docker_uri_base_file='./results/docker/URI_BASE'
+                        mkdir -p $(dirname ${docker_uri_base_file})
+                        echo ${URI_BASE} > "${docker_uri_base_file}"
+                            sg docker -c "
+                                docker login -u '${USER}' -p '${PASS}'
+                                export IMAGE=\$URI_BASE
+
+                                for app in "pgo-apiserver" "pgo-event" "pgo-rmdata" "pgo-scheduler" "postgres-operator" "pgo-deployer"; do
+                                    docker pull perconalab/percona-postgresql-operator:$GIT_BRANCH-$GIT_PREV_SHORT_COMMIT-\\${app}
+                                    docker tag perconalab/percona-postgresql-operator:$GIT_BRANCH-$GIT_PREV_SHORT_COMMIT-\\${app} \\${IMAGE}-\\${app}
+                                    docker push \\${IMAGE}-\\${app}
+                                done
+                                docker logout
+                            "
+                        sudo rm -rf ./build
+                    '''
+                }
+                stash includes: 'results/docker/URI_BASE', name: 'URI_BASE'
+                archiveArtifacts 'results/docker/URI_BASE'
+            }
+        }
         stage('Build docker image') {
             when {
-                expression {
-                    !skipBranchBulds
```

Comment:
we should not remove it because we will have:
https://cloud.cd.percona.com/job/cloud-pg-operator/job/K8SPG-38-data-migration-test/11/
we need to skip branch build

Reply:
done